src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs

```diff
@@ -99,6 +99,33 @@ public bool IsAbsoluteTemplate
             };
         }
 
+        /// <summary>
+        /// Combines the prefix and route template for an attribute route.
+        /// </summary>
+        /// <param name="prefix">The prefix.</param>
+        /// <param name="template">The route template.</param>
+        /// <returns>The combined pattern.</returns>
+        public static string CombineTemplates(string prefix, string template)
```

Comment:
@rynowak, how do you feel about these APIs? I'm not particularly vested in this, but this was my interpretation of not duplicating code per https://github.com/aspnet/Mvc/pull/5450#discussion_r85363884